pkg/datapath/linux/config/config.go

```diff
@@ -140,6 +132,48 @@ func (h *HeaderfileWriter) WriteClusterConfig(w io.Writer) error {
 	return fw.Flush()
 }
 
+// HeaderfileReader is a wrapper type which implements datapath.ConfigReader.
+// It manages reading and validation of datapath program headerfiles.
+type HeaderfileReader struct{}
+
+func NewHeaderfileReader() *HeaderfileReader {
+	return &HeaderfileReader{}
+}
+
+// ValdateClusterConfig reads an existing cluster-wide options config file and
+// validates it against the given configuration.
+func (h *HeaderfileReader) ValidateClusterConfig(r io.Reader, cfg datapath.ClusterConfiguration) error {
+	cDefinesMap := cfg.GetOptions()
+
+	for key, currentVal := range h.ReadConfigMacros(r) {
+		if newVal, ok := cDefinesMap[key]; ok && newVal != currentVal {
+			return fmt.Errorf("cluster_config.h cannot be changed. %s was defined as %s, but new agent configuration would change it to %v", key, currentVal, newVal)
+		}
+	}
+
+	return nil
+}
```

Comment:
Yeah, I agree that it would only occur for re-installations changing that parameter, but that would still be a possibility (especially because the block is to prevent changing it while cilium is installed, so there would be no other way). And having to remove that file from all the nodes doesn't seem to be that user-friendly either. 

I'm not sure how other possible "unsafe" parameter changes have been handled in the past, but my concern is just avoiding that the prevention turns out to be more problematic than the original issue itself (I mean, changing that parameter will break the connectivity in the whole cluster, but that could be even fine while performing tests).